core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java

```diff
@@ -296,50 +294,50 @@ private ManagedKafkaProducer createManagedKafkaProducer(final StreamRegistryConf
     }
 
     private StreamValidator createStreamValidator(final StreamRegistryConfiguration configuration, final Environment environment) {
-        Preconditions.checkState(regionDao != null, "regionDao cannot be null.");
+        Preconditions.checkState(regionService != null, "regionService cannot be null.");
 
         final Client httpClient = new JerseyClientBuilder(environment).using(configuration.getHttpClient()).using(environment)
             .build("remoteStateStoreHttpClient");
 
         log.info("Connection Timeout:{}", configuration.getHttpClient().getConnectionTimeout());
         log.info("Connection Request Timeout:{}", configuration.getHttpClient().getConnectionRequestTimeout());
 
-        StreamValidator streamValidator = loadValidator(configuration, httpClient, regionDao);
+        StreamValidator streamValidator = loadValidator(configuration, httpClient, regionService);
         return streamValidator;
     }
 
-    private void initDao(final StreamRegistryConfiguration configuration, final Environment environment, ManagedKStreams managedKStreams,
-        ManagedKafkaProducer managedKafkaProducer) {
+    private void initServiceAndDao(final StreamRegistryConfiguration configuration, final Environment environment, ManagedKStreams managedKStreams,
+                                   ManagedKafkaProducer managedKafkaProducer) {
 
         Preconditions.checkState(managedKStreams != null, "managedKStreams cannot be null.");
         Preconditions.checkState(managedKafkaProducer != null, "managedKafkaProducer cannot be null.");
         Preconditions.checkState(infraManager != null, "infraManager cannot be null.");
 
         String env = configuration.getEnv();
 
-        regionDao = new RegionDaoImpl(env, infraManager);
-        clusterDao = new ClusterDaoImpl(infraManager);
+        regionService = new RegionServiceImpl(env, infraManager);
+        clusterService = new ClusterServiceImpl(infraManager);
 
         KafkaManager kafkaManager = new KafkaManagerImpl();
         SchemaManager schemaManager = loadSchemaManager(configuration);
         StreamValidator streamValidator = createStreamValidator(configuration, environment);
 
-        streamDao = new StreamDaoImpl(managedKafkaProducer, managedKStreams, env, regionDao, clusterDao, infraManager, streamValidator, schemaManager,
-            kafkaManager);
-        producerDao = new ProducerDaoImpl(managedKafkaProducer, managedKStreams, env, regionDao, clusterDao, infraManager);
-        consumerDao = new ConsumerDaoImpl(managedKafkaProducer, managedKStreams, env, regionDao, clusterDao, infraManager);
+        streamDao = new StreamDaoImpl(managedKafkaProducer, managedKStreams);
```

Comment:
Created a new StreamDaoImpl with managedKafkaProducer and managedKStreams to extract out the calls.

```
    void upsertStream(AvroStreamKey key, AvroStream stream);
    Pair<AvroStreamKey, Optional<AvroStream>> getStream(String streamName);
    KeyValueIterator<AvroStreamKey, AvroStream> getAllStreams();
```